File: app/src/main/java/com/simulando/joaopaulodribeiro/simulando/Callbacks.java
package com.simulando.joaopaulodribeiro.simulando;

import android.view.View;

import com.simulando.joaopaulodribeiro.simulando.model.simulates.AnswerTest;
import com.simulando.joaopaulodribeiro.simulando.model.simulates.Test;

/**
 * Created by joao.paulo.d.ribeiro on 01/10/2017.
 */

public interface Callbacks {

    interface OnNotifySimulatesHomeFragmentListener {
        void onNotifySimulatesHomeFragment(int position);
    }

    interface OnMillisUntilFinishedListener {
        void onMillisUntilFinished(long millisUntilFinished);
    }

    interface OnQuestionAnsweredListener {
        void OnQuestionAnswered(AnswerTest answerTest, int fragmentPosition);
    }

    interface OnViewPagerChangeFragmentListener {
        void OnViewPagerChangeFragment();
    }

}

File: app/src/main/java/com/simulando/joaopaulodribeiro/simulando/Callbacks.kt
package com.simulando.joaopaulodribeiro.simulando

import android.view.View

import com.simulando.joaopaulodribeiro.simulando.model.simulates.AnswerTest
import com.simulando.joaopaulodribeiro.simulando.model.simulates.Test

/**
 * Created by joao.paulo.d.ribeiro on 01/10/2017.
 */

interface Callbacks {

    interface OnNotifySimulatesHomeFragmentListener {
        fun onNotifySimulatesHomeFragment(position: Int)
    }

    interface OnMillisUntilFinishedListener {
        fun onMillisUntilFinished(millisUntilFinished: Long)
    }

    interface OnTimerCountDownFinishedListener {
        fun onTimerCountDownFinished()
    }

    interface OnQuestionAnsweredListener {
        fun OnQuestionAnswered(answerTest: AnswerTest, fragmentPosition: Int)
    }

    interface OnViewPagerChangeFragmentListener {
        fun OnViewPagerChangeFragment()
    }

}
